index.js

```diff
@@ -0,0 +1,65 @@
+const tilesContainer = document.querySelector(".tiles");
+
+//memotest images
+
+const feelingsImgLink = {
+    happy: "images/feelings/happy.png",
+    sad: "images/feelings/sad.png", 
+    angry: "images/feelings/angry.png",
+    hungry: "images/feelings/hungry.png",
+    thirsty: "images/feelings/thirsty.png",
+    scared: "images/feelings/scared.png",
+}
+
+const feelingsImages = {}; //object that contains images to append
+const feelingsPairs = []; //this is to shuffle the cards internally
+
+for (const [feeling, src] of Object.entries(feelingsImgLink)) {
+const img = document.createElement("img");
+img.src = src;
+feelingsImages[feeling] = img;
+feelingsPairs.push(feeling, feeling); //this is to have each feeling twice in an array that will be shuffled later
+}
+
+//create each tile
+function buildTile (feeling) {
+    const card = document.createElement("img");
+    card.src = "images/square_backside_pattern.png"
+
+    card.classList.add("card");
+    card.setAttribute("data-feeling", feeling) // this is to know which feeling is asigned to the card, whether it is revealed or not
+    tilesContainer.appendChild(card); //maybe we need to get this out of the function & put it in the for loop
+
+    let unrevealedCard = true; //toggle between states
+    card.addEventListener("click", () => {
+        if (unrevealedCard) {
+            card.src = feelingsImgLink[feeling];
+            unrevealedCard = false;
+            
+        } else {
+            card.src = "images/square_backside_pattern.png";
+            unrevealedCard = true;
+            
+        }
+
+        //add animation no matter the click
+        card.classList.add("animation"); //i got this animation from a css animation builder 
+
+        //remove it so the class is added and the animation triggered every time there's a click. the delay time matches the duration of the animation (0.75s)
+        setTimeout(() => {
+            card.classList.remove("animation")
+        }, 750);
+    })
+}
+
+//tiles shuffling & access to images
+const tilesCount = feelingsPairs.length;
+
+for (let i=0; i<tilesCount; i++) { //tilesCount has to be here before the feelingsPairs.length changes
```

Comment:
I know the forEach array method, it's a higher-order function like .map( ) and .filter( ), correct? I know that .map and .filter also apply a callback function just as .forEach, and they also return arrays. I guess forEach only executes the function inside, but doesn't return an array.

What I don't get though is why you used -0.5 in your randomSortComparison function. I can also see you're using .sort as a kind of attribute(?) of the feelingsPairs array. I don't know what that is, could you please tell me?

Reply:
Yes, they are! They are powerful tools for your code.

I gave you more details in private messages, but for the record:

Math.random() generates a number between 0 and 1 as a float. By subtracting 0.5, you get a number between -0.5 and 0.5, which means a 50% chance of a negative number or a positive number. This number will be used as an argument for the sort() method.

sort() orders the array. It takes two arguments (usually a and b) and returns:

A negative number:  a should appear before b.
A positive number:  a should appear after b.
Zero: they are equal in terms of order.

[Array.prototype.sort()](https://developer.mozilla.org/es/docs/Web/JavaScript/Reference/Global_Objects/Array/sort)

Reply:
Hey Nico! Thanks! After some research I finally got it!
Actually, I found out that it can be done in one line only: `feelingsPairs.sort(() => Math.random() - 0.5); `